Add documentations for the release process of Helm charts (#723)
ray-project/kuberay: docs/release/helm-chart.md

# Helm charts release 

We host all Helm charts on [kuberay-helm](https://github.com/ray-project/kuberay-helm). This document describes the process for release managers to release Helm charts.

## The end-to-end workflow

> ### Step1: Update version in Chart.yaml

Reviewer: ```suggestion
### Step 1: Update version in Chart.yaml
```

Please update the value of `version` in [ray-cluster/Chart.yaml](https://github.com/ray-project/kuberay/blob/master/helm-chart/ray-cluster/Chart.yaml), [kuberay-operator/Chart.yaml](https://github.com/ray-project/kuberay/blob/master/helm-chart/kuberay-operator/Chart.yaml), and [kuberay-apiserver/Chart.yaml](https://github.com/ray-project/kuberay/blob/master/helm-chart/kuberay-apiserver/Chart.yaml) to the new release version (e.g. 0.4.0).

> ### Step2: Copy the helm-chart directory from kuberay to kuberay-helm

Reviewer: ```suggestion
### Step 2: Copy the helm-chart directory from kuberay to kuberay-helm
```

In [kuberay-helm CI](https://github.com/ray-project/kuberay-helm/blob/main/.github/workflows/chart-release.yaml), `helm/chart-releaser-action` will create releases for all charts in the directory `helm-chart` and update `index.yaml` in the [gh-pages](https://github.com/ray-project/kuberay-helm/tree/gh-pages) branch when the PR is merged into `main`. Note that `index.yaml` is necessary when you run the command `helm repo add`. I recommend removing the `helm-chart` directory in the kuberay-helm repository and creating a new one by copying from the kuberay repository.

> ### Step3: Check the correctness

Reviewer: ```suggestion
### Step 3: Check the correctness
```

When the PR is merged into `main`, the releases and `index.yaml` will be generated. You can check the correctness by:

* Check whether the [releases](https://github.com/ray-project/kuberay-helm/releases) are created as expectation.
* Check whether [index.yaml](https://github.com/ray-project/kuberay-helm/blob/gh-pages/index.yaml) exists or not.
* Check whether [index.yaml](https://github.com/ray-project/kuberay-helm/blob/gh-pages/index.yaml) has metadata of all releases, including old versions.
* Check the creation/update time of all releases and `index.yaml` to ensure they are updated.

* Install charts from Helm repository.
    ```sh
    helm repo add kuberay https://ray-project.github.io/kuberay-helm/

    # List all charts
    helm search repo kuberay

    # Install charts
    helm install kuberay-operator kuberay/kuberay-operator
    helm install kuberay-apiserver kuberay/kuberay-apiserver
    helm install ray-cluster kuberay/ray-cluster   
    ```

> ## Delete the existing releases

Reviewer: Why do we need to delete existing releases?

Should we maintain Helm release version for each KubeRay release?

Author: > Why do we need to delete existing releases?

Because the `chart-releaser` will only be triggered when the PR is merged into `main`, maintainers may want to revert some wrong PRs. For example, a release manager of 0.4.0 forgets to update `version` from 0.3.0 to 0.4.0 in Chart.yaml, so the existing 0.3.0 releases and `index.yaml` update. In this case, we need to follow the "Delete the existing releases" section to revert.

> Should we maintain Helm release version for each KubeRay release?

Yes

Reviewer: Ok, makes sense -- could you clarify that deleting charts is only for such situations?

Reviewer: Actually, I think it already sufficiently conveys that deleting things is not recommended.
